plugin/evm/gossiper.go

```diff
@@ -84,86 +84,87 @@ func (vm *VM) newPushGossiper() Gossiper {
+	maxTxs int,
+	maxAcctTxs int,
+) types.Transactions {
+	var (
+		stxs     = types.NewTransactionsByPriceAndNonce(n.signer, txs, baseFee)
+		statuses = make(map[common.Address]*addrStatus)
+		queued   = make([]*types.Transaction, 0, maxTxs)
+	)
+
+	// Iterate over possible transactions until there are none left or we have
+	// hit the regossip target.
+	for len(queued) < maxTxs {
+		next := stxs.Peek()
+		if next == nil {
+			break
 		}
 
-		// Ensure any transactions regossiped are immediately executable
-		var (
-			currentNonce = state.GetNonce(addr)
-			tx           *types.Transaction
-		)
-		for _, accountTx := range accountTxs {
-			// The tx pool may be out of sync with current state, so we iterate
-			// through the account transactions until we get to one that is
-			// executable.
-			if accountTx.Nonce() == currentNonce {
-				tx = accountTx
-				break
-			}
-			// There may be gaps in the tx pool and we could jump past the nonce we'd
-			// like to execute.
-			if accountTx.Nonce() > currentNonce {
-				break
+		sender, _ := types.Sender(n.signer, next)
+		status, ok := statuses[sender]
+		if !ok {
+			status = &addrStatus{
+				nonce: state.GetNonce(sender),
 			}
-		}
-		if tx == nil {
-			continue
+			statuses[sender] = status
 		}
 
-		// Don't try to regossip a transaction too frequently
-		if time.Since(tx.FirstSeen()) < n.config.TxRegossipFrequency.Duration {
+		// The tx pool may be out of sync with current state, so we iterate
+		// through the account transactions until we get to one that is
+		// executable.
+		switch {
+		case next.Nonce() < status.nonce:
+			stxs.Shift()
 			continue
-		}
-
-		// Ensure the fee the transaction pays is valid at tip
-		wrapped, err := types.NewTxWithMinerFee(tx, baseFee)
-		if err != nil {
-			log.Debug(
-				"not queuing tx for regossip",
-				"tx", tx.Hash(),
-				"err", err,
-			)
+		case next.Nonce() > status.nonce, time.Since(next.FirstSeen()) < regossipFrequency.Duration,
+			status.txsAdded >= maxAcctTxs:
+			stxs.Pop()
 			continue
 		}
-
-		heads = append(heads, wrapped)
-	}
-	heap.Init(&heads)
-
-	// Add up to [maxTxs] transactions to be gossiped
-	queued := make([]*types.Transaction, 0, maxTxs)
-	for len(heads) > 0 && len(queued) < maxTxs {
-		tx := heads[0].Tx
-		queued = append(queued, tx)
-		heap.Pop(&heads)
+		queued = append(queued, next)
+		status.nonce++
+		status.txsAdded++
+		stxs.Shift()
 	}
-
 	return queued
 }
 
-// queueRegossipTxs finds the best transactions in the mempool and adds up to
-// [TxRegossipMaxSize] of them to [txsToGossip].
+// queueRegossipTxs finds the best non-priority transactions in the mempool and adds up to
+// [RegossipMaxTxs] of them to [txsToGossip].
 func (n *pushGossiper) queueRegossipTxs() types.Transactions {
 	// Fetch all pending transactions
 	pending := n.txPool.Pending(true)
 
+	// Remove all priority transactions
+	for _, account := range n.config.PriorityRegossipAddresses {
+		delete(pending, account)
+	}
+
```

Comment:
Can we remove this and allow priority transactions to be included in both sets of gossip?